src/deploy/functions/cache/hash.ts

```diff
@@ -0,0 +1,69 @@
+import { readFile } from "node:fs/promises";
+import * as crypto from "crypto";
+import { Backend, Endpoint } from "../backend";
+import { getSecretVersions } from "../../../functions/secrets";
+
+/**
+ * Generates a hash from the environment variables of a {@link Backend}.
+ * @param backend Backend of a set of functions
+ */
+export function getEnvironmentVariablesHash(backend: Backend): string {
+  const hash = crypto.createHash("sha256");
+
+  // Hash the contents of the dotenv variables
+  const hasEnvironmentVariables = !!Object.keys(backend.environmentVariables).length;
+  if (hasEnvironmentVariables) {
+    hash.update(JSON.stringify(backend.environmentVariables));
+  }
+
+  return hash.digest("hex");
+}
```

Comment:
I think we should get environment variable hash for an endpoint and not from the backend. It's possible that an endpoint might have different set of environment variables from another endpoint in the same backend. For example, you can go to Pantheon and manually add an env var entry for a function.

Reply:
Spoke offline. The understanding we came to was that in instances of wantBackend, `backend.environmentVariables === endpoint.environmentVariables`.

So in the interest of conserving the # of hashing requests, I'm keeping it on the `Backend` level and allowing it to be shared with the hashing of multiple Endpoints.